x-pack/plugins/transform/public/app/sections/transform_management/components/edit_transform_flyout/edit_transform_ingest_pipeline.tsx

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useEffect, useState, type FC } from 'react';
+
+import { EuiComboBox, EuiFormRow } from '@elastic/eui';
+
+import { i18n } from '@kbn/i18n';
+
+import { isEsIngestPipelines } from '../../../../../../common/api_schemas/type_guards';
+
+import { useApi } from '../../../../hooks/use_api';
+
+import { EditTransformFlyoutFormTextInput } from './edit_transform_flyout_form_text_input';
+import { useEditTransformFlyout } from './use_edit_transform_flyout';
+
+export const EditTransformIngestPipeline: FC = () => {
+  const { errorMessages, value } = useEditTransformFlyout('destinationIngestPipeline');
+  const { formField } = useEditTransformFlyout('actions');
+
+  const api = useApi();
+
+  const [ingestPipelineNames, setIngestPipelineNames] = useState<string[]>([]);
+
+  useEffect(function fetchPipelinesOnMount() {
+    let unmounted = false;
+
+    async function getIngestPipelineNames() {
+      const ingestPipelines = await api.getEsIngestPipelines();
+
+      if (!unmounted && isEsIngestPipelines(ingestPipelines)) {
+        setIngestPipelineNames(ingestPipelines.map(({ name }) => name));
+      }
+    }
+
+    getIngestPipelineNames();
+
+    return () => {
+      unmounted = true;
+    };
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  return (
+    <>
+      {
+        // If the list of ingest pipelines is not available
+        // gracefully defaults to text input
+        ingestPipelineNames ? (
```

Comment:
You initialise `ingestPipelineNames` with `[]`, hence this check is always truthy. Perhaps we should check the array length instead 
```suggestion
        ingestPipelineNames.length > 0 ? (
```

Reply:
Good catch! I fixed that check in [ee405ae](https://github.com/elastic/kibana/pull/149670/commits/ee405ae0be5a27ebbb1bf6bee766d0c5f8e47b71). Additionally I improved the loading behavior of the dropdown in 
269ca09.